theappraisallane/blackbook-client
# [SUB-2044] Added Blackbook namespace

**WARNING: This is a breaking change, so bumped version to 2.0.0**

**- What I did**
I added a `Blackbook` namespace to this library. Nothing was changed regarding implementation/functionality.

**- How I did it**
Some clarification of why this was changed, we have a Drilldown endpoint (being implemented) in vin-decoder-api, and I'm having issues when wanting to use a DrilldownService class, which is already being declared in this library. After researching a bit I found that libraries should be namespaced so their declared constants/names don't collide with clients using them. [Here](https://github.com/savonrb/savon/blob/master/lib/savon.rb)'s an example.

**- How to verify it**
Tests should pass. Use the library from a client that has a class with the same name of one of the components of this library, and it should work just fine after this change.

**- Description for the changelog**
[SUB-2044] Added Blackbook namespace

**- A picture of a cute animal (not mandatory but encouraged)**
![](https://i.ytimg.com/vi/Xw-lOzRenzA/hqdefault.jpg)

[SUB-2044]: https://theappraisallane.atlassian.net/browse/SUB-2044
